### PR TITLE
Fix functorch test selection logic

### PR DESCRIPTION
**Pull request by huydhn**
I realize that `run_test.py` doesn't take into account functorch test selection logic at the moment, for example `python test/run_test.py --functorch -i functorch/test_ops --verbose` stills run all functorch tests

### CODE REVIEW

**Review by clee2000 — APPROVED** [review]

File: test/run_test.py [hunk 1]
```diff
@@ -1018,7 +1018,9 @@ def get_selected_tests(options):
         )
 
     if options.functorch:
-        selected_tests = FUNCTORCH_TESTS
+        selected_tests = list(
+            filter(lambda test_name: test_name in FUNCTORCH_TESTS, selected_tests)
+        )
```
Comment (malfet) [inline test/run_test.py:1023]: Use list comprehensions
```suggestion
        selected_tests = [tname for tname in selected_test if tname in FUNCTORCH_TESTS]
```

**Review by malfet — APPROVED** [review]

File: test/run_test.py [hunk 2]
```diff
@@ -1018,7 +1018,7 @@ def get_selected_tests(options):
         )
 
     if options.functorch:
-        selected_tests = FUNCTORCH_TESTS
+        selected_tests = [tname for tname in selected_test if tname in FUNCTORCH_TESTS]
```
Comment (malfet) [inline test/run_test.py:1021]:
```suggestion
        selected_tests = [tname for tname in selected_tests if tname in FUNCTORCH_TESTS]
```

**Pull request closed without merging**